controllers/job.go

```diff
@@ -65,16 +65,21 @@ func unpackContainer(id, containerImage, pullImage string, pullOptions buildv1al
 	}
 }
 
-func createImageContainer(containerImage string, pushOptions buildv1alpha1.Push) v1.Container {
+func createImageContainer(containerImage string, artifact buildv1alpha1.OSArtifact) v1.Container {
+	imageName := "dontpush" // No image was defined, use a dummy one to let luet work
+	if artifact.Spec.PushOptions.ImageName != "" {
+		imageName = artifact.Spec.PushOptions.ImageName
+	}
 	return v1.Container{
 		ImagePullPolicy: v1.PullAlways,
 		Name:            "create-image",
 		Image:           containerImage,
 		Command:         []string{"/bin/bash", "-cxe"},
 		Args: []string{
 			fmt.Sprintf(
-				"tar -czvpf test.tar -C /rootfs . && luet util pack %s test.tar image.tar && mv image.tar /artifacts",
-				pushOptions.ImageName,
+				"tar -czvpf test.tar -C /rootfs . && luet util pack %[1]s test.tar %[2]s.tar && chmod +r %[2]s.tar && mv %[2]s.tar /artifacts",
+				imageName,
```

Comment:
I think we should accept an image name generically. For pushing I'd leave seaparately. We could infer a default at least from the spec name to avoid any user input

Reply:
There is some redundancy in `OSArtifactSpec.ImageName` and `OSArtifactSpec.PushOptions.ImageName`. I don't see why they would ever be different. Maybe we should consolidate them to one option and make it mandatory. We can let the user decide to use a dummy name for an image if they don't care to push it.

My preference would be to remove the `ImageName` from the `PushOptions` since it's needed even when we don't push the image.

Reply:
Because there are 2 separate images involved here: one used where to pull from (our current `ImageName`, which might be a kairos-core image) and the one that is the result of our customization - if we do any - at rootfs level, hence why you see two of them here.

Reply:
gotcha! Updated the code now.